src/services/ui/src/pages/form/medicaid-form.tsx

```diff
@@ -0,0 +1,349 @@
+    const fileMetaData: NonNullable<OneMacTransform["attachments"]> = [];
+
+    const presignedUrls: Promise<PreSignedURL>[] = uploadKeys
+      .filter((key) => data.attachments[key] !== undefined)
+      .map(() =>
+        API.post("os", "/getUploadUrl", {
+          body: {},
+        })
+      );
+    const loadPresignedUrls = await Promise.all(presignedUrls);
+
+    uploadKeys
+      .filter((key) => data.attachments[key] !== undefined)
+      .forEach((uploadKey, index) => {
+        const attachmenListObject = attachmentList?.find(
+          (item) => item[0] === uploadKey
+        );
+        const title = attachmenListObject ? attachmenListObject[1] : "Other";
+        const fileGroup = data.attachments[uploadKey] as File[];
+
+        // upload all files in this group and track there name
+        for (const file of fileGroup) {
+          uploadedFiles.push(
+            fetch(loadPresignedUrls[index].url, {
+              body: file,
+              method: "PUT",
+            })
+          );
+
+          fileMetaData.push({
+            key: loadPresignedUrls[index].key,
+            filename: file.name,
+            title: title,
+            bucket: loadPresignedUrls[index].bucket,
+            uploadDate: Date.now(),
+          });
+        }
+      });
+
+    await Promise.all(uploadedFiles);
+
+    const dataToSubmit: OneMacTransform & {
+      state: string;
+      proposedEffectiveDate: number;
+      authority: string;
+    } = {
+      id: data.id,
+      additionalInformation: data?.additionalInformation ?? null,
+      attachments: fileMetaData,
+      origin: "micro",
+      authority: "medicaid spa",
+      raiResponses: [],
+      submitterEmail: user?.user?.email ?? "N/A",
+      submitterName:
+        `${user?.user?.given_name} ${user?.user?.family_name}` ?? "N/A",
+      proposedEffectiveDate: data.proposedEffectiveDate.getTime(),
+      state: data.id.split("-")[0],
+    };
+
+    let submissionResponse;
+    try {
+      submissionResponse = await API.post("os", "/submit", {
+        body: dataToSubmit,
+      });
+      console.log(submissionResponse);
+      setModalChildren(
+        <div className="flex flex-col gap-2 items-center text-center">
+          <div className="max-w-md p-4 rounded-lg shadow-lg">
+            <div className="font-bold">Submission Success!</div>
+            <p>
+              Your submission was accepted.
+              <br />
+              Please be aware that it may take up to a minute for your
+              submission to show in the Dashboard.
+            </p>
+          </div>
+          <I.Button
+            type="button"
+            variant="outline"
+            onClick={() => navigate(ROUTES.DASHBOARD)}
+          >
+            Go to Dashboard
+          </I.Button>
+        </div>
+      );
+    } catch (err) {
+      console.log(err);
+      setModalChildren(
+        <div className="flex flex-col gap-2 items-center text-center">
```

Comment:
Extract this so you can one-line it; `setModal(<ErrorModalContent />)`

Reply:
I don't see support for validating the presence of required attachments, nor validating non-duplicates for the case of `cms179`. The schema seems to have them all as nullish arrays.